tests/CreateComposerZipArtifacts.php

```diff
@@ -0,0 +1,259 @@
+<?php
+
+/**
+ * This file is part of the Dealerdirect PHP_CodeSniffer Standards
+ * Composer Installer Plugin package.
+ *
+ * @copyright 2022 PHPCodeSniffer Composer Installer Contributors
+ * @license MIT
+ */
+
+namespace Dealerdirect\Composer\Plugin\Installers\PHPCodeSniffer\Tests;
+
+use DirectoryIterator;
+use RecursiveCallbackFilterIterator;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use RuntimeException;
+use SplFileInfo;
+use ZipArchive;
+
+/**
+ * Helper class to create the Composer artifact zip packages for use in the tests.
+ *
+ * This class is used in the test `bootstrap.php` file.
+ */
+class CreateComposerZipArtifacts
+{
+    /**
+     * Default version number to use for fixture packages.
+     *
+     * This version number can be overruled by setting the `version` key in the fixture package
+     * `composer.json` file or by passing the version number explicitly to the
+     * `createZipArtifact()` method.
+     *
+     * @var string
+     */
+    const FIXTURE_VERSION = '1.0.0';
+
+    /**
+     * The full path to the directory to place the zipped artifacts in (including trailing slash).
+     *
+     * @var string
+     */
+    private $artifactDir;
+
+    /**
+     * List of file subpaths which should be excluded from the zip archives.
+     *
+     * Note: no need to list files starting with a `.` as those will always be filtered out.
+     *
+     * @link https://www.php.net/manual/en/recursivedirectoryiterator.getsubpathname.php
+     *
+     * @var array
+     */
+    private $excludedFiles = array(
+        'composer.lock'    => 'composer.lock',
+        'phpcs.xml.dist'   => 'phpcs.xml.dist',
+        'phpunit.xml.dist' => 'phpunit.xml.dist',
+        'phpunit.xml'      => 'phpunit.xml',
+    );
+
+    /**
+     * List of file extensions for files which should be excluded from the zip archives.
+     *
+     * @var array
+     */
+    private $excludedExtensions = array(
+        'md'   => 'md',
+        'bak'  => 'bak',
+        'orig' => 'orig',
+    );
+
+    /**
+     * List of top-level directories which should be excluded from the zip archives.
+     *
+     * Note: no need to list directories starting with a `.` as those will always be filtered out.
+     *
+     * @var array
+     */
+    private $excludedDirs = array(
+        'bin'    => 'bin',
+        'build'  => 'build', // PHPUnit code coverage directory.
+        'tests'  => 'tests',
+        'vendor' => 'vendor',
+    );
+
+    /**
+     * Constructor.
+     *
+     * @param string $artifactDir Full path to the directory to place the sipped artifacts in.
+     */
+    public function __construct($artifactDir)
+    {
+        // Make sure the directory has a trailing slash.
+        if (substr($artifactDir, -1) !== '/') {
+            $artifactDir .= '/';
+        }
+
+        $this->artifactDir = $artifactDir;
```

Comment:
To my mind, this would be more readable:
```php
        // Make sure the directory has a trailing slash.
        $this->artifactDir = rtrim($artifactDir, '/') . '/';
```